tls/s2n_config.c

```diff
@@ -1071,3 +1072,36 @@ int s2n_config_set_recv_multi_record(struct s2n_config *config, bool enabled)
 
     return S2N_SUCCESS;
 }
+
+/* Indicates if the connection should attempt to enable kTLS. */
+int s2n_config_ktls_enable(struct s2n_config *config, s2n_ktls_mode ktls_mode)
+{
+    POSIX_ENSURE_REF(config);
+
+    config->ktls_mode_requested = ktls_mode;
+    return S2N_SUCCESS;
+}
+
+/* Enabling Duplex mode means that both Send and Recv are also requested.
+ *
+ * config                   ktls_mode                                   requested
+ * ->ktls_mode_requested
+ * ------------------------------------------------------------------------------
+ * S2N_KTLS_MODE_DUPLEX     S2N_KTLS_MODE_DUPLEX                        true
+ * S2N_KTLS_MODE_SEND       S2N_KTLS_MODE_SEND | S2N_KTLS_MODE_DUPLEX   true
+ * S2N_KTLS_MODE_RECV       S2N_KTLS_MODE_RECV | S2N_KTLS_MODE_DUPLEX   true
+ * S2N_KTLS_MODE_DISABLED   S2N_KTLS_MODE_DISABLED                      true
+ */
+bool s2n_config_is_ktls_requested(struct s2n_config *config, s2n_ktls_mode ktls_mode)
+{
+    POSIX_ENSURE_REF(config);
```

Comment:
This is exactly why the return value should not be `bool`. If this check fails, it'll return a `-1`, which will be cast to `true`.